includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -302,6 +302,15 @@ public function create_and_confirm_setup_intent( $payment_method_id, $customer_i
 
 		if ( WC_Payments_Features::is_sepa_enabled() ) {
 			$request['payment_method_types'] = [ Payment_Method::CARD, Payment_Method::SEPA ];
+			$request['mandate_data']         = [
+				'customer_acceptance' => [
+					'type'   => 'online',
+					'online' => [
+						'ip_address' => WC_Geolocation::get_ip_address(),
+						'user_agent' => $_SERVER['HTTP_USER_AGENT'] ?? $this->user_agent, //phpcs:ignore WordPress.Security.ValidatedSanitizedInput
+					],
+				],
+			];
```

Comment:
Maybe we should add a test in `tests/unit/wc-payment-api/test-class-wc-payments-api-client.php` just in case. Thoughts?

Reply:
I agree.
We don't have any tests either for the function `create_and_confirm_intention` where we also added code for SEPA. I will take a look at that one too.

Reply:
@harriswong I added a test for `create_and_confirm_setup_intent`
The feature flag makes it sort of temporary because we will have to update the test when we remove it.
The check for the flag is on a static function and I had to update the option on the DB for the test.
Because of the flag, I didn't add another test for `create_and_confirm_intention`.